ml/src/test/scala/ai/h2o/sparkling/ml/models/H2OMOJOModelTestSuite.scala

```diff
@@ -34,6 +39,80 @@ class H2OMOJOModelTestSuite extends FunSuite with SharedH2OTestContext with Matc
+        objectStream.readObject().asInstanceOf[H2OMOJOModel]
+      }
+    }
+
+    def assertDataframesOnMOJOModels(expected: H2OMOJOModel, deserialized: H2OMOJOModel): Unit = {
+      deserialized shouldNot be(null)
+
+      val expectedFeatureImportances = expected.getFeatureImportances()
+      val expectedScoringHistory = expected.getScoringHistory()
+      val expectedTrainingMetrics = expected.getTrainingMetricsObject().asInstanceOf[H2OBinomialMetrics]
+      val expectedThresholdsAndMetricScores = expectedTrainingMetrics.getThresholdsAndMetricScores()
+      val expectedMaxCriteriaAndMetricScores = expectedTrainingMetrics.getMaxCriteriaAndMetricScores()
+      val expectedGainsLiftTable = expectedTrainingMetrics.getGainsLiftTable()
+      val expectedConfusionMatrix = expectedTrainingMetrics.getConfusionMatrix()
+
+      val featureImportances = deserialized.getFeatureImportances()
+      val scoringHistory = deserialized.getScoringHistory()
+      val trainingMetrics = expected.getTrainingMetricsObject().asInstanceOf[H2OBinomialMetrics]
+      val thresholdsAndMetricScores = trainingMetrics.getThresholdsAndMetricScores()
+      val maxCriteriaAndMetricScores = trainingMetrics.getMaxCriteriaAndMetricScores()
+      val gainsLiftTable = expectedTrainingMetrics.getGainsLiftTable()
+      val confusionMatrix = expectedTrainingMetrics.getConfusionMatrix()
+
+      featureImportances shouldNot be(null)
+      scoringHistory shouldNot be(null)
+      thresholdsAndMetricScores shouldNot be(null)
+      maxCriteriaAndMetricScores shouldNot be(null)
+      gainsLiftTable shouldNot be(null)
+      confusionMatrix shouldNot be(null)
+
+      TestUtils.assertDataFramesAreIdentical(expectedFeatureImportances, featureImportances)
+      TestUtils.assertDataFramesAreIdentical(expectedScoringHistory, scoringHistory)
+      TestUtils.assertDataFramesAreIdentical(expectedThresholdsAndMetricScores, thresholdsAndMetricScores)
+      TestUtils.assertDataFramesAreIdentical(expectedMaxCriteriaAndMetricScores, maxCriteriaAndMetricScores)
+      TestUtils.assertDataFramesAreIdentical(expectedGainsLiftTable, gainsLiftTable)
+      TestUtils.assertDataFramesAreIdentical(expectedConfusionMatrix, confusionMatrix)
+    }
+
+    val expectedCVModels = model.getCrossValidationModels()
+    val expectedCVScoringHistory = model.getCrossValidationModelsScoringHistory()
+    val expectedCrossValidationSummary = model.getCrossValidationMetricsSummary()
+
+    val crossValidationSummary = deserializedModel.getCrossValidationMetricsSummary()
+    val deserializedCVModels = deserializedModel.getCrossValidationModels()
+    val deserializedCVScoringHistory = deserializedModel.getCrossValidationModelsScoringHistory()
+    TestUtils.assertDataFramesAreIdentical(expectedCrossValidationSummary, crossValidationSummary)
+    assertDataframesOnMOJOModels(model, deserializedModel)
+    var i = 0
+    while (i < expectedCVModels.length) {
```

Comment:
`for (i <- expectedCVModels.indices)`
or
`for (i <- 0 until expectedCVModels.length)`

would be more "scala-ish"

Reply:
Yeah, here it makes sense. The number of cv models is small. On a bigger collection I would prefer `while` loop because of the significant performance benefits.